### ISSUE TITLE
PATH problem when setting locale due to no login shell

### ISSUE THREAD

**Issue by fghsgh** [2bdkid/debian-rpi4-arm64-guide#3]
By default, when entering the chroot, you do not get a login shell. This means environment variables do not get initialized properly. In my case, this was problematic, because it was missing `/sbin` from the PATH which was necessary for e.g. setting the locale. I fixed it by running `chroot /mnt su -` (where `su -` opens a login shell).

**Comment by 2bdkid** [comment]
Was the issue with dpkg-reconfigure?

**Comment by fghsgh** [comment]
Yes, but apt also complained. The `locale-gen` program was unavailable.

**Comment by 2bdkid** [comment]
I can't seem to reproduce your exact error. I did experience a weird segmentation fault tho. In this chroot `locales` is the only additional package installed.

```
[brady@fedora rpi]$ sudo chroot ./debian/ /bin/bash
[sudo] password for brady: 
root@fedora:/# dpkg-reconfigure locales
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
/usr/bin/locale: Cannot set LC_CTYPE to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_MESSAGES to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_ALL to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_CTYPE to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_MESSAGES to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_ALL to default locale: No such file or directory
Generating locales (this might take a while)...
  en_US.UTF-8...Segmentation fault (core dumped)
 done
Generation complete.
*** update-locale: Error: invalid locale settings:  LANG=en_US.UTF-8
root@fedora:/# dpkg-reconfigure locales
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
/usr/bin/locale: Cannot set LC_CTYPE to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_MESSAGES to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_ALL to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_CTYPE to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_MESSAGES to default locale: No such file or directory
/usr/bin/locale: Cannot set LC_ALL to default locale: No such file or directory
Generating locales (this might take a while)...
  en_US.UTF-8... done
Generation complete.
```

**Comment by 2bdkid** [comment]
Further, `/sbin` is included in my chroot `PATH` without the login shell. What's your host OS?

```
root@fedora:/# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/var/lib/snapd/snap/bin
```

**Comment by fghsgh** [comment]
I am running Arch Linux, where `/sbin` (as well as `/bin`) are symlinks to `/usr/bin` by default. The problem was that it couldn't find the `locale` program. Anyway, it's never a good idea to go to a different environment without setting the environment variables accordingly, and that is exactly what opening a login shell does.

**Comment by 2bdkid** [comment]
Fixed with de4d7d33ef2f61d6aa25c2352fb9858d0996cfba
